### DISCUSSION TITLE
May I ask what software did you use to debug?

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#631] [Q&A] [answered]
I'm running your example by using `./SeisSol_Release_dhsw_6_elastic parameters.par` and `SeisSol_Release_dhsw_6_elastic` is your compiled executable, I can't debug and set breakpoints to track the progress of the program while running. If I want to debug and trace the program by setting breakpoints, what compiler should I use? I'm going to use `visual studio code` now, but I don't know how to read the parameters file. Last time I saw someone recommend gcc+gdb to me, but I'm not good at gdb. I like to use `visual studio code`, a tool with a graphical interface, no Knowing if there are any recommended compilers to use, thanks a lot.
![QQ截图20220724155201](https://user-images.githubusercontent.com/104721358/180638258-e896e4ed-ac12-4a35-b03d-2fdf57d76333.png)

**Comment by sebwolf-de** [comment, marked as answer]
Hi, first you should build the `Debug` version of SeisSol, which will build SeisSol without compiler optimizations and with debug flags. Debugging software is not an easy task and you should read some tutorials about it. Working with debuggers is not something, which I can explain to you in a short github discussion.
As I have said earlier, I use gcc+gdb to debug SeisSol and it works nicely for me.

**Reply by smallriver666** [reply]
> Hi, first you should build the `Debug` version of SeisSol, which will build SeisSol without compiler optimizations and with debug flags. Debugging software is not an easy task and you should read some tutorials about it. Working with debuggers is not something, which I can explain to you in a short github discussion. As I have said earlier, I use gcc+gdb to debug SeisSol and it works nicely for me.

Thanks for your reply, I would like to ask where can I find the `Debug` version of SeisSol ? Also, is there a recommended complete gcc+gdb tutorial? I haven't studied it before, so I don't know which tutorial is more suitable, including using the input example to complete the trace of the program and how to read the parameter file if it is not compiled into an executable program?   thank you a lot.

**Reply by sebwolf-de** [reply]
`cmake .. -DCMAKE_BUILD_TYPE=Debug`

**Reply by sebwolf-de** [reply]
You should really try to find a suitable tutorial for gdb on your own. There are many many tutorials if you google or search on youtube. I do not know your background and your expertise, so it doen't make any sense, if I point you to a specific tutorial.

**Reply by smallriver666** [reply]
> You should really try to find a suitable tutorial for gdb on your own. There are many many tutorials if you google or search on youtube. I do not know your background and your expertise, so it doen't make any sense, if I point you to a specific tutorial.

Ok, thank you very much for your enthusiastic help. My undergraduate major has nothing to do with this. I just started learning, so I have zero foundation, so sometimes I may ask some stupid questions, please bear with me . Finally, thank you again for your help, I will try to learn it myself first.
